hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/append/AppendWriteFunction.java

```diff
@@ -112,7 +112,7 @@ protected void sendBootstrapEvent() {
     int attemptId = getRuntimeContext().getAttemptNumber();
     if (attemptId > 0) {
       // either a partial or global failover, reuses the current inflight instant
-      if (this.currentInstant != null) {
+      if (this.currentInstant != null && !metaClient.getActiveTimeline().filterCompletedInstants().containsInstant(currentInstant)) {
```

Comment:
Make sure currentInstant is truly inflight

Reply:
It is possible.   
writeClient.commit  is successful,  but the  log `"Commit instant [{}] success!` doesn't exist. Something happened when  this.ckpMetadata.commitInstant(instant).

After flink job restart, the  WriteFunction's instant is `20231114121907373`